### PR TITLE
Improved list and feed errors

### PR DESCRIPTION
**Pull request by pfrazee**
- Improve the error state of feedgens to appear less frightening
- Fix the error state of profile lists
- Unsave lists when deleted

## Feedgen errors

|||
|-|-|
|CleanShot 2023-11-02 at 15 09 01@2x|CleanShot 2023-11-02 at 15 09 07@2x|

## List error state

CleanShot 2023-11-02 at 13 47 55@2x

### CODE REVIEW

File: src/state/models/feeds/posts.ts [hunk 1]
```diff
@@ -25,6 +25,17 @@ import {MergeFeedAPI} from 'lib/api/feed/merge'
...
+type FeedType = 'home' | 'following' | 'author' | 'custom' | 'likes' | 'list'
+
+export enum KnownError {
+  FeedgenDoesNotExist,
+  FeedgenMisconfigured,
+  FeedgenBadResponse,
+  FeedgenOffline,
+  FeedgenUnknown,
+  Unknown,
+}
```
Comment (estrattonbailey) [inline src/state/models/feeds/posts.ts:37]: Not a hill I'd die on, but I think we should only use string enums. These get evaluated to numbers, which makes rearranging difficult, etc.

```
export enum KnownError {
  FeedgenDoesNotExist = 'FeedgenDoesNotExist'
  ...
}
```

Works 100% of the time, very clear.

File: src/view/com/posts/FeedErrorMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}: {
+  feed: PostsFeedModel
+  onPressTryAgain: () => void
+}) {
+  if (
+    typeof feed.knownError === 'undefined' ||
+    feed.knownError === KnownError.Unknown
+  ) {
+    return (
+      <ErrorMessage message={feed.error} onPressTryAgain={onPressTryAgain} />
```
Comment (estrattonbailey) [inline src/view/com/posts/FeedErrorMessage.tsx:39]: Hehe so we fall back to the red error?

**Review by estrattonbailey — APPROVED** [review]
LGTM, no blockers

File: src/view/com/posts/FeedErrorMessage.tsx (hunk 2 above)
Comment (pfrazee) [reply src/view/com/posts/FeedErrorMessage.tsx:39]: Hah ya but only for non-feedgens

**Pull request merged**